pkg/web/sources.go

```diff
@@ -132,11 +139,53 @@ func showStats(ctx *gin.Context) (bool, bool) {
 	return parse(ctx, strconv.ParseBool, st)
 }
 
+func showHealth(ctx *gin.Context) (bool, bool) {
+	st := ctx.Query("health")
+	if st == "" {
+		return false, true
+	}
+	return parse(ctx, strconv.ParseBool, st)
+}
+
+func (c *Controller) isHealthy(ctx context.Context, isSource bool, id int64) (*bool, error) {
```

Comment:
Return a simple bool instead of *bool here because the NOT EXISTS query always returns a bool value. The pointers upwards the stack could therefore also replaced by a simple bool.